### PR TITLE
impl(storage): new helpers for CRC32C computation

### PR DESCRIPTION
**Pull request by coryan**
I am adding new functions to compute CRC32C checksums. When using recent versions of Abseil (>= 20230125) these functions can concatenate the checksum of two buffers. I am planning to use this in the GCS+gRPC plugin, where each `Write()` message includes a CRC32C and we also need to send the end-to-end checksum of all the `Write()` messages.

Part of the work #11060

- - -
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/11061)

### CODE REVIEW

File: google/cloud/storage/internal/crc32c_benchmark.cc [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+//   L1 Data 32 KiB (x64)
+//   L1 Instruction 32 KiB (x64)
+//   L2 Unified 512 KiB (x64)
+//   L3 Unified 16384 KiB (x16)
+// Load Average: 1.88, 2.61, 6.87
+// ----------------------------------------------------------------------
+// Benchmark                            Time             CPU   Iterations
+// ----------------------------------------------------------------------
+// BM_Crc32cDuplicateNonAbseil   25520759 ns     25520833 ns           28
+// BM_Crc32cDuplicate            24168074 ns     24168122 ns           28
```
Comment (dbolduc) [inline google/cloud/storage/internal/crc32c_benchmark.cc:36]: Now that I see the implementation, I am not surprised that they take 2x as long.

File: google/cloud/storage/BUILD.bazel [hunk 2]
```diff
@@ -165,3 +166,16 @@ load(":storage_client_grpc_unit_tests.bzl", "storage_client_grpc_unit_tests")
         "@com_google_googletest//:gtest_main",
     ],
 ) for test in storage_client_grpc_unit_tests]
+
+load(":storage_client_benchmarks.bzl", "storage_client_benchmarks")
```
[the remark this answers is not in the document]
Comment (coryan) [reply google/cloud/storage/BUILD.bazel:170]: Those are "integration" benchmarks.  We have something similar in Spanner with little "unit" benchmarks at the same level as unit tests, and "integration" benchmarks in a special directory.

File: google/cloud/storage/google_cloud_cpp_storage.cmake [hunk 3]
```diff
@@ -559,6 +564,26 @@ if (BUILD_TESTING)
...
+        google_cloud_cpp_add_executable(target "storage" "${fname}")
+        add_test(NAME ${target} COMMAND ${target})
+        target_link_libraries(
+            ${target} PRIVATE google-cloud-cpp::storage storage_client_testing
+                              benchmark::benchmark_main)
+        google_cloud_cpp_add_common_options(${target})
+    endforeach ()
+
     add_subdirectory(tests)
     add_subdirectory(benchmarks)
```
[the remark this answers is not in the document]
Comment (coryan) [reply google/cloud/storage/google_cloud_cpp_storage.cmake:588]: Ack

File: google/cloud/storage/internal/crc32c_benchmark.cc (hunk 1 above)
Comment (coryan) [reply google/cloud/storage/internal/crc32c_benchmark.cc:36]: Ack.

**Review by dbolduc — APPROVED** [review]

**Pull request merged**